File: src/main/java/com/wanted/preonboarding/ticket/config/JpaConfig.java
package com.wanted.preonboarding.ticket.config;

import com.querydsl.jpa.impl.JPAQueryFactory;
import jakarta.persistence.EntityManager;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.jpa.repository.config.EnableJpaAuditing;

@EnableJpaAuditing
@Configuration
public class JpaConfig {
    @Bean
    public JPAQueryFactory jpaQueryFactory(EntityManager entityManager) {
        return new JPAQueryFactory(entityManager);
    }
}

File: src/main/kotlin/com/wanted/preonboarding/ticket/infra/entity/config/JpaConfig.kt
package com.wanted.preonboarding.ticket.infra.entity.config

import org.springframework.boot.autoconfigure.domain.EntityScan
import org.springframework.context.annotation.Configuration
import org.springframework.data.jpa.repository.config.EnableJpaRepositories
import org.springframework.transaction.annotation.EnableTransactionManagement

@Configuration
@EnableTransactionManagement
@EntityScan(basePackages = ["com.wanted.preonboarding.ticket.infra"])
@EnableJpaRepositories(basePackages = ["com.wanted.preonboarding.ticket.infra"])
class JpaConfig
